core/src/snapshot_package.rs

```diff
@@ -76,7 +71,13 @@ impl SnapshotPackagerService {
         let tar_gz = tempfile::Builder::new()
             .prefix("new_state")
             .suffix(".tgz")
-            .tempfile()?;
+            .tempfile_in(
+                snapshot_package
+                    .tar_output_file
+                    .parent()
```

Comment:
Should we `fs::create_dir_all()` this path at some point prior?

Reply:
Right now those paths are created in `validator/main.rs`, but it's probably not a bad idea to guarantee creation here as well.

Reply:
Right now those paths are created in validator/main.rs, but it's probably not a bad idea to guarantee creation here as well.